a3p-integration/proposals/z:acceptance/test-lib/kread.js

```diff
@@ -0,0 +1,346 @@
+
+  const id = `KREAd-unequip-all-items-acceptance-test`;
+  const body = {
+    method: 'executeOffer',
+    offer: {
+      id,
+      invitationSpec: {
+        source: 'agoricContract',
+        instancePath: ['kread'],
+        callPipe: [['makeUnequipAllInvitation', []]],
+      },
+      proposal: {
+        give: {
+          CharacterKey1: {
+            brand: brands.KREAdCHARACTER,
+            value: makeCopyBag([[kreadCharacter, 1n]]),
+          },
+        },
+        want: {
+          CharacterKey2: {
+            brand: brands.KREAdCHARACTER,
+            value: makeCopyBag([[kreadCharacter2, 1n]]),
+          },
+        },
+      },
+    },
+  };
+
+  return JSON.stringify(marshaller.toCapData(harden(body)));
+};
+
+const buyItemOffer = async () => {
+  const children = await getMarketItemsChildren();
+  const marketItem = await getMarketItem(children[0]);
+
+  const itemPrice =
+    marketItem.askingPrice.value +
+    marketItem.platformFee.value +
+    marketItem.royalty.value;
+
+  const id = `KREAd-buy-item-acceptance-test`;
+  const body = {
+    method: 'executeOffer',
+    offer: {
+      id,
+      invitationSpec: {
+        source: 'agoricContract',
+        instancePath: ['kread'],
+        callPipe: [['makeBuyItemInvitation', []]],
+      },
+      offerArgs: { entryId: marketItem.id },
+      proposal: {
+        give: {
+          Price: {
+            brand: brands.IST,
+            value: itemPrice,
+          },
+        },
+        want: {
+          Item: {
+            brand: brands.KREAdITEM,
+            value: makeCopyBag([[marketItem.asset, 1n]]),
+          },
+        },
+      },
+    },
+  };
+
+  return JSON.stringify(marshaller.toCapData(harden(body)));
+};
+
+const sellItemOffer = async address => {
+  const kreadItem = await getBalanceFromPurse(address, 'item');
+  if (!kreadItem) {
+    throw new Error('Item not found on user purse');
+  }
+
+  const id = `KREAd-sell-item-acceptance-test`;
+  const body = {
+    method: 'executeOffer',
+    offer: {
+      id,
+      invitationSpec: {
+        source: 'agoricContract',
+        instancePath: ['kread'],
+        callPipe: [['makeSellItemInvitation', []]],
+      },
+      proposal: {
+        give: {
+          Item: {
+            brand: brands.KREAdITEM,
+            value: makeCopyBag([[kreadItem, 1n]]),
+          },
+        },
+        want: {
+          Price: {
+            brand: brands.IST,
+            value: 5n * ISTunit,
+          },
+        },
+      },
+    },
+  };
+
+  return JSON.stringify(marshaller.toCapData(harden(body)));
+};
+
+const buyCharacterOffer = async () => {
+  const charactersMarket = await getMarketCharactersChildren();
+  const path = `:published.kread.market-characters.${charactersMarket[0]}`;
+  const rawCharacterData = await agoric.follow('-lF', path, '-o', 'text');
+  const marketCharacter = marshaller.fromCapData(JSON.parse(rawCharacterData));
```

Comment:
In the effort to keep the new helper function `getAssetAmount` agnostic to the asset provided as an argument, I left the query to Vstorage unchanged and used marketCharacter as the argument. 
Does this approach work for you?

Reply:
I'm fine with passing `marketCharacter` to `totalAssetPriceAmount`, but I'd still prefer that there be a function for the 4 lines above. (perhaps `getMarketCharacterFromVstorage()`).
is `0` a parameter saying which child, or is it always `':published.kread.market-characters.${charactersMarket[0]}'`?

Reply:
I addressed the requested change by implementing the new getMarketCharacterFromVstorage helper function.

In this case, there is no need to provide the market-characters children node dynamically because, for the test case built, we only need to ensure that a character is available on the market. 
For that reason, I am using the static child index 0.